### PR TITLE
Include Rust YJIT in libruby_static.a

### PR DESCRIPTION
**Pull request by XrXr**
Prior to this bigdecimal had linking issue which it detected during
mkmf configuration and that made it not build properly. This lets us
try to boot `make test-all`.

This works by combining two archive libraries with ar(1). I also tried a
couple of other approaches. `cargo rustc -- --emit obj` outputs an
object file but at a path that's hard to find and considered internal to
Cargo according to the manual. `ld -r --whole-archive` seems to be an
alternative, but unfortunately is unavailable with the linker on macOS.

### CODE REVIEW

**Comment by XrXr** [comment]
Second commit gets us closer to booting `make test-all`.

File: yjit/src/core.rs [hunk 1]
```diff
@@ -618,7 +619,7 @@ pub fn take_version_list(blockid: BlockId) -> VersionList
     if insn_idx >= payload.version_map.len() {
         VersionList::default()
     } else {
-        payload.version_map.remove(insn_idx)
+        mem::take(&mut payload.version_map[insn_idx])
```
Comment (maximecb) [inline yjit/src/core.rs:622]: Are those things not equivalent?

**Comment by maximecb** [comment]
Can you explain how the library merging works and why it's necessary? :O

File: yjit/src/core.rs (hunk 1 above)
Comment (XrXr) [reply yjit/src/core.rs:622]: `Vec::remove` shifts the elements after it which we don't want.

Comment (maximecb) [reply yjit/src/core.rs:622]: Aaaaah!

**Comment by XrXr** [comment]
`libruby_static.a` is installed for C extension gems to consume with `make install`. Archives are pretty much a ZIP file with all the `.o` files and and an index; we make the C `.o` files depend on symbols that the Rust object files export so we need them all in one archive. stdlib has some extensions too and those are built when doing `make` using similar tools to external extension gems.

The merging is essentially unzipping the Rust output then repacks the object files into libruby. Some directory fnagling required since ar(1) only extracts to the current working directory.

**Pull request merged**